src/StackYaml/Transformations/UpdateExtraDeps.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE OverloadedStrings #-}
+module StackYaml.Transformations.UpdateExtraDeps
+  ( transformation
+  ) where
+
+import Control.Applicative (many, (<|>))
+import Control.Lens
+import Data.Aeson.Lens
+import Data.List           (intercalate, intersperse)
+import Data.Monoid         ((<>))
+import Data.Text           (Text, pack, unpack)
+import Data.Yaml           (Value)
+
+import qualified Data.Attoparsec.Text  as Atto
+import qualified Distribution.Package  as Cabal
+import qualified Distribution.PackDeps as PackDeps
+import qualified Distribution.Version  as Cabal
+
+transformation :: Value -> IO Value
+transformation =
+    traverseOf (key "extra-deps" . values . _String) updateExtraDep
+  where
+    updateExtraDep :: Text -> IO Text
+    updateExtraDep dep =
+      either
+        (\err -> dep <$ putStrLn err)
+        (fmap pPrintExtraDep . getNewest)
+        (Atto.parseOnly (pExtraDep <* Atto.endOfInput) dep)
+
+pExtraDep :: Atto.Parser ExtraDep
```

Comment:
Actually, should we just `regex-applicative-text` here. Would be much clearer then attoparsec (and can be more resilient as well)